--- visualization/src/pcl_visualizer.cpp.orig	2018-09-07 10:25:29.229299121 +0800
+++ visualization/src/pcl_visualizer.cpp	2018-09-07 10:25:40.280522332 +0800
@@ -100,7 +100,7 @@
 #include <pcl/visualization/pcl_visualizer.h>
 #include <pcl/visualization/common/common.h>
 #include <pcl/common/time.h>
-#include <boost/uuid/sha1.hpp>
+#include <boost/uuid/detail/sha1.hpp>
 #include <boost/filesystem.hpp>
 #include <pcl/console/parse.h>
 
